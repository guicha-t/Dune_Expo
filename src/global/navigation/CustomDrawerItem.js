import PropTypes from 'prop-types';
import React, {Component} from 'react';
import {ScrollView, Text, View, StyleSheet, TouchableOpacity, Image} from 'react-native';

import Store from './../store/Store'

export default class CustomDrawerItems extends Component {

  renderElement(){
   if(Store.TypeUser == 2)
        return <TouchableOpacity onPress={() => this.props.navigation.navigate('AddUserDirector')}>
          <View style={styles.button}>
            <View style={styles.labelContainer}>
              <Text style={styles.label}>Ajouter un professeur</Text>
            </View>
          </View>
        </TouchableOpacity>;
   return null;
  }

  render () {
    return (
      <View style={styles.container}>
        <ScrollView>

          <TouchableOpacity onPress={() => this.props.navigation.navigate('Dashboard')}>
            <View style={styles.button}>
              <View style={styles.labelContainer}>
                <Text style={styles.label}>Dashboard</Text>
              </View>
            </View>
          </TouchableOpacity>

          <TouchableOpacity onPress={() => this.props.navigation.navigate('StudentList')}>
            <View style={styles.button}>
              <View style={styles.labelContainer}>
                <Text style={styles.label}>Trombinoscope</Text>
              </View>
            </View>
          </TouchableOpacity>

          <TouchableOpacity onPress={() => this.props.navigation.navigate('GamesList')}>
            <View style={styles.button}>
              <View style={styles.labelContainer}>
                <Text style={styles.label}>Logith√®que</Text>
              </View>
            </View>
          </TouchableOpacity>

          <TouchableOpacity onPress={() => this.props.navigation.navigate('UserDemands')}>
            <View style={styles.button}>
              <View style={styles.labelContainer}>
                <Text style={styles.label}>Notification</Text>
              </View>
            </View>
          </TouchableOpacity>


          {this.renderElement() }

          <TouchableOpacity onPress={() => this.props.navigation.navigate('Profil')}>
            <View style={styles.button}>
              <View style={styles.labelContainer}>
                <Text style={styles.label}>Profil</Text>
              </View>
            </View>
          </TouchableOpacity>

          <TouchableOpacity onPress={() => this.props.navigation.navigate('QRCode')}>
            <View style={styles.button}>
              <View style={styles.labelContainer}>
                <Text style={styles.label}>Scan QR-Code</Text>
              </View>
            </View>
          </TouchableOpacity>

          <TouchableOpacity onPress={() => this.props.navigation.navigate('TermsOfUse')}>
            <View style={styles.button}>
              <View style={styles.labelContainer}>
                <Text style={styles.label}>CGU</Text>
              </View>
            </View>
          </TouchableOpacity>

          <TouchableOpacity onPress={() => this.props.navigation.navigate('ContactForm')}>
            <View style={styles.button}>
              <View style={styles.labelContainer}>
                <Text style={styles.label}>Nous contacter</Text>
              </View>
            </View>
          </TouchableOpacity>


        </ScrollView>
      </View>
    );
  }
}

CustomDrawerItems.propTypes = {
  navigation: PropTypes.object
};

const styles = StyleSheet.create({
  container: {
      flex: 1
    },
    button: {
      flexDirection: 'row',
      height: 42,
    },
    icon: {
      flex: 0.2,
      justifyContent: 'center',
      alignItems: 'center',
    },
    labelContainer: {
      flex: 0.6,
      justifyContent: 'center',
      alignItems: 'flex-start',
      paddingLeft: 20,
      paddingTop: 5,
    },
    label: {
      color:'#363453',
      fontSize: 16,
    },

});
