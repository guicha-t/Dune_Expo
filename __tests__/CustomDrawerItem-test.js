import CustomDrawerItem from './../src/global/navigation/CustomDrawerItem';
import 'isomorphic-fetch';
import React from 'react';

import renderer from 'react-test-renderer'

jest.useFakeTimers();

test('render CustomDrawerItem correctly', () => {

    const tree = renderer.create(<CustomDrawerItem />).toJSON();
    expect(tree).toMatchSnapshot();
});