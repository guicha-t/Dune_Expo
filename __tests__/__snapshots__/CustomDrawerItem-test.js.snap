// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render CustomDrawerItem correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "borderBottomWidth": 1,
            "borderColor": "white",
            "height": 40,
            "justifyContent": "flex-end",
            "opacity": 0.5,
            "paddingRight": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          ACCUEIL
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 42,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 0.6,
                "justifyContent": "center",
                "paddingLeft": 20,
                "paddingTop": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Dashboard
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "borderBottomWidth": 1,
            "borderColor": "white",
            "height": 40,
            "justifyContent": "flex-end",
            "opacity": 0.5,
            "paddingRight": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          STATISTIQUES
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 42,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 0.6,
                "justifyContent": "center",
                "paddingLeft": 20,
                "paddingTop": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Historique
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 42,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 0.6,
                "justifyContent": "center",
                "paddingLeft": 20,
                "paddingTop": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Trombinoscope
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "borderBottomWidth": 1,
            "borderColor": "white",
            "height": 40,
            "justifyContent": "flex-end",
            "opacity": 0.5,
            "paddingRight": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          APPLICATIONS
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 42,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 0.6,
                "justifyContent": "center",
                "paddingLeft": 20,
                "paddingTop": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Logith√®que
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "borderBottomWidth": 1,
            "borderColor": "white",
            "height": 40,
            "justifyContent": "flex-end",
            "opacity": 0.5,
            "paddingRight": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          TABLE
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 42,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 0.6,
                "justifyContent": "center",
                "paddingLeft": 20,
                "paddingTop": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Scan QR-Code
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "borderBottomWidth": 1,
            "borderColor": "white",
            "height": 40,
            "justifyContent": "flex-end",
            "opacity": 0.5,
            "paddingRight": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          DIVERS
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 42,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 0.6,
                "justifyContent": "center",
                "paddingLeft": 20,
                "paddingTop": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Profil
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 42,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 0.6,
                "justifyContent": "center",
                "paddingLeft": 20,
                "paddingTop": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              CGU
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 42,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 0.6,
                "justifyContent": "center",
                "paddingLeft": 20,
                "paddingTop": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Nous contacter
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 42,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 0.6,
                "justifyContent": "center",
                "paddingLeft": 20,
                "paddingTop": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Guide d'utilisation
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 40,
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
